components/cdc/src/endpoint.rs

```diff
@@ -1435,8 +1435,10 @@ impl<T: 'static + RaftStoreRouter<E>, E: KvEngine> Runnable for Endpoint<T, E> {
                 cb();
             }
             Task::TxnExtra(txn_extra) => {
-                for (k, v) in txn_extra.old_values {
-                    self.old_value_cache.cache.insert(k, v);
+                if !self.capture_regions.is_empty() {
+                    for (k, v) in txn_extra.old_values {
```

Comment:
Can we stop taking old_values in txn layer when there is no downstream registered?